### PR TITLE
Add CMEK to Google Secret Manager

### PR DESCRIPTION
**Pull request by SKAUL05**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/8917

If this PR is for Terraform, I acknowledge that I have:

 

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran make test and make lint to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

Release Note Template for Downstream PRs (will be copied)

```release-note:enhancement
secret-manager: added support for setting a CMEK on `google_secret_manager_secret` 
```

### CODE REVIEW

**Comment by SKAUL05** [comment]
@c2thorn   Have you reviewed it yet?

**Comment by SKAUL05** [comment]
> /gcbrun

I think build didn't run or did it?

**Comment by c2thorn** [comment]
> @c2thorn I think build didn't run or did it?

Silly mistake on my part, I thought we were in https://github.com/GoogleCloudPlatform/magic-modules 🤣 

Heads up, Magic Modules is our upstream repository that generates code for this resource. Changes made here will get overridden. I've added the field and your test in https://github.com/GoogleCloudPlatform/magic-modules/pull/4752 and it looks mostly the same with a few minor differences (ordering). But everything seems to work.

File: website/docs/r/secret_manager_secret.html.markdown [hunk 1]
```diff
@@ -100,6 +100,18 @@ The `replicas` block supports:
   (Required)
   The canonical IDs of the location to replicate data. For example: "us-east1".
 
+* `custom_managed_encryption` -
```
Comment (c2thorn) [inline website/docs/r/secret_manager_secret.html.markdown:103]: This ended up needing to be `customer_managed_encryption`

**Comment by c2thorn** [comment]
The upstream PR will now get reviewed. If it's all the same to you, we can close this PR and let the upstream PR maintain any further changes if needed (I've added you as a co-author).

**Comment by SKAUL05** [comment]
> The upstream PR will now get reviewed. If it's all the same to you, we can close this PR and let the upstream PR maintain any further changes if needed (I've added you as a co-author).

Works for me!

**Pull request closed without merging**